plugins/statistics.koplugin/main.lua

```diff
@@ -1447,42 +1449,81 @@ function ReaderStatistics:getCurrentStat()
+
+    local current_page
+    local total_pages
+    local page_progress_string
+    local percent_read
+    if (self.view.document:hasHiddenFlows()) then
+        local flow = self.view.document:getPageFlow(self.view.state.page)
+        current_page = self.view.document:getPageNumberInFlow(self.view.state.page)
+        total_pages = self.view.document:getTotalPagesInFlow(flow)
+        percent_read = Math.round(100*current_page/total_pages)
+        if flow == 0 then
+            page_progress_string = ("%d // %d (%d%%)"):format(current_page, total_pages, percent_read)
+        else
+            page_progress_string = ("[%d / %d]%d (%d%%)"):format(current_page, total_pages, flow, percent_read)
+        end
+    else
+        current_page = self.view.state.page
+        total_pages = self.data.pages
+        percent_read = Math.round(100*current_page/total_pages)
+        page_progress_string = ("%d / %d (%d%%)"):format(current_page, total_pages, percent_read)
+    end
+
+    local first_open_days_ago = math.floor(tonumber(now_ts - first_open)/86400)
+    local time_to_read = current_page and ((total_pages - current_page) * self.avg_time) or 0
     local estimate_days_to_read = math.ceil(time_to_read/(book_read_time/tonumber(total_days)))
-    local estimate_end_of_read_date = os.date("%Y-%m-%d", tonumber(now_ts + estimate_days_to_read * 86400))
+    local estimate_end_of_read_date = datetime.secondsToDate(tonumber(now_ts + estimate_days_to_read * 86400), true)
     local estimates_valid = time_to_read > 0 -- above values could be 'nan' and 'nil'
     local user_duration_format = G_reader_settings:readSetting("duration_format", "classic")
+    local avg_page_time_string = datetime.secondsToClockDuration(user_duration_format, self.avg_time, false, true)
+    local avg_day_time_string = datetime.secondsToClockDuration(user_duration_format, book_read_time/tonumber(total_days), false, true)
+    local time_to_read_string = estimates_valid and datetime.secondsToClockDuration(user_duration_format, time_to_read, false, true) or _("N/A")
+
+    -- Use more_arrow to indicate that an option shows another view
+    -- Use "*" to indicate that an option will show an info message
+    local more_arrow = BD.mirroredUILayout() and "◂" or "▸"
+
+    local estimated_popup = function()
+        UIManager:show(InfoMessage:new{
+            text = T(N_("%1 page (%2%) of the book left.", "%1 pages (%2%) of the book left.", total_pages - current_page), total_pages - current_page, 100 - percent_read) ..
+                "\n\n" .. T(_("At the current rate of %1 per page, the book will be finished in %2 of reading time."), avg_page_time_string, time_to_read_string) ..
+                "\n\n" .. T(N_("At the current rate of %1 read per day, the book will be finished in %2 typical day of reading.", "At the current rate of %1 read per day, the book will be finished in %2 typical days of reading.", estimate_days_to_read), avg_day_time_string, estimate_days_to_read),
```

Comment:
@Frenzie These ones!

Reply:
That first item in N_ should be "1 page", not "%1 page". It'll resolve to the same thing in the final output but don't bother people with that. ;-)

Also I'd rather say "1 page left" without "of the book" since people often read things that aren't books like papers, newspapers, individual long-form articles, etc.

There's a discrepancy between "per page" and "read per day"

Reply:
Oops, missed that, fixed. Pushing soon

Taking out book, we'll probably end up with:

> There are 337 pages (78%) left to read.
> At the current rate of 36 pages per day, there is 6h37m21s of reading time left.
> At the current rate of 21m57s per day, there are 27 days of reading left.

For the other one, I put "3h67m22s **read** per day" because we discriminate between read time and total time spent, and wanted to clarify that this is only considering read time, not total spent time. But I'll still take out "read" because it should be fairly easy to glean from the rest of the sentence and the fact that we're in this very reading-centered message in the first place—it's not a message for measuring your dilly dallying

Reply:
What's total spent time?

Reply:
Whatever I knew about the statistics plugin I forgot long ago, other than that the calendar view looks really, really cool. ;-)

Anyway, so if I understand correctly total spent time is like how my playtime of a game on GOG or Steam basically tracks the *maximum* amount of time I may have spent on it, but it also counts when I left it paused for an hour.

Additionally iirc a page isn't counted if it doesn't meet a minimum time, but it would add a few seconds to total time spent.

Reply:
Yes indeed

I realized the grammatical form would be "there are 6h37m21s", and we'd probably need ngettext twice in one string. So to avoid it... how's "that is 6h37m21s"?

>There are 337 pages (78%) left to read.
>At the current rate of 36 pages per day, that is 6h37m21s of reading time left.
>At the current rate of 21m57s per day, that is 27 days of reading left.

Reply:
Actually that sounds awkward. Maybe this:

> There are 337 pages (78%) left to read.
> At the current rate of 1m20s per page, **that will take** 6h37m21s of reading time.
> At the current rate of 21m57s per day, **that will take** 27 days of reading.

Reply:
They're both grammatical because those are different sentences. Amounts and quantities use singular verbs and pronouns.

There are 337 pages left to read. That is 6 hours of reading time left.
NOT There are 337 pages left to read. <del>Those are 6 hours of reading time left.</del>

But if you say "there are" then you're not talking about the quantity but saying something by itself, so it's also grammatical.

But yes, this is a weakness of the ngettext design we haven't quite figured out how best to deal with yet, other than to avoid the scenario when reasonably possible.

Reply:
> At the current rate of 1m20s per page, that will take 6h37m21s of reading time.
At the current rate of 21m57s per day, that will take 27 days of reading.

Drop the `of reading`.

Reply:
I guess I was thinking of this case: "There is 3s of reading time left." That wouldn't be grammatical

Reply:
>There are 337 pages (78%) left to read.
>At the current rate of 1m20s per page, that will take 6h37m21s of reading time.
>At the current rate of 21m57s per day, that will take 27 days.

Reply:
> I guess I was thinking of this case: "There is 3s of reading time left." That didn't sound grammatical to me cuz I saw it as "There is 3 seconds of reading time left", which to me is ungrammatical. Unless we do 0m3s, I actually haven't checked, in which case it would be fine... ha

That would be ungrammatical because it's very different from "There are 10 words remaining. That is 3 s of reading time left." Where `that is` refers to the quantity of `10 words`.

> At the current rate of 1m20s per page, that will take 6h37m21s of reading time.

I meant the same for this one.

Reply:
In my dialect/idiolect, I would treat amounts less differently from any other phrase. Though I've heard a lot of people say "There's 3 seconds left". Seems like a [gray area](https://qr.ae/prp96q) / transitional / dialectal thing.

>I meant the same for this one.

I'm not sure about that one, we need to keep the distinction of reading time and time spent. Idk. @poire-z?

Reply:
> Sorry @Frenzie, we all too often forget you are new to the project :)
> 
> ![image](https://user-images.githubusercontent.com/24273478/207703982-53aa4d67-841f-4e9b-a53d-1ff067b88777.png)
> 
> First one is total sum of durations. Second one is "-- This is the query used to compute the total time spent reading distinct pages of the book, capped at self.settings.max_sec per distinct page." (Not really sure of how they end up differing, even looking at the code :) don't we cap them at insertion time ? @NiLuJe knows.)
> 
> (May be these one too could get a popup to explain what they are.)

The first one is essentially the amount of time spent with the book open (e.g., including back and forth), while the second only accounts for a page *once*, which each page capped at the value set by the user for that.

Reply:
@melyux 
> In my dialect/idiolect, I would treat amounts less differently from any other phrase. Though I've heard a lot of people say "There's 3 seconds left". Seems like a [gray area](https://qr.ae/prp96q) / transitional / dialectal thing.

The gray area linked doesn't refer to the same thing.

As I said:
> There are 337 pages left to read. That is 6 hours of reading time left.
NOT There are 337 pages left to read. <del>Those are 6 hours of reading time left.</del>

The link speaks of a different scenario:

There are 337 pages and 10 words left to read. That adds up to 6 hours of reading time left.
NOT? There are 337 pages and 10 words left to read. Those add up to 6 hours of reading time left.

The way I see it, you'd still have to go for "that," because "337 pages and 10 words" is one and the same quantity, identical to "337 pages" all by its lonesome self. Antecedents joined by *and* usually take plural pronouns. But not when both antecedents refer to a single thing, so it still takes a singular pronoun because one of them would.

It'd be different if you said there are 337 pages and 10 chicken bones left to read.

Reply:
> I'm not sure about that one, we need to keep the distinction of reading time and time spent. It might imply that we're calculating a total time taking into account time wastes.

Well, can't say anything about the grammar.

> There are 337 pages (78%) left to read.
At the current rate of 1m20s per page, that will take 6h37m21s of reading time.
At the current rate of 21m57s per day, that will take 27 days.

I like that one, mentionning once "of reading time".
But I'll be also ok with (may be this final wording needs less to have "reading time" than some earlier wording when I felt it was needed).

> There are 337 pages (78%) left to read.
At the current rate of 1m20s per page, that will take 6h37m21s.
At the current rate of 21m57s per day, that will take 27 days.

Reply:
Yeah I think it fits fine here. Let's do it

Reply:
Both of the final contenders are fine by me, but the "of reading time" feels like a bit of unnecessary wordiness. Not a big deal.

Reply:
I thought I committed the version without "of reading time" but apparently didn't. But if we're fine with it, I'll keep it in that one place